src/drivers/PdfViewer/PdfViewer.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect, useRef } from 'react';
+import * as pdfjsLib from 'pdfjs-dist/build/pdf';
+import PropTypes from 'prop-types';
+import PdfPage from './PdfPage';
+
+export default function PDFViewer({ filePath, disableVisibilityCheck }) {
+  const [error, setError] = useState(null);
+  const canvasRef = useRef(null);
+  const [pdf, setPdf] = useState(null);
+  const [percent, setPercent] = useState(0);
+
+  useEffect(() => {
+    if (canvasRef?.current) {
+      const loadingTask = pdfjsLib.getDocument(filePath);
+
+      loadingTask.onProgress = (progress) => {
+        const newPercent = (progress.loaded / progress.total) * 100;
+        setPercent(newPercent);
+      };
+
+      loadingTask.promise.then((result) => setPdf(result)).catch(setError);
+    }
+  }, [filePath, canvasRef?.current]);
+
+  const renderPages = () => {
```

Comment:
Why are these 2 functions defined like this? `renderPage` and `renderLoading`? Let's just write them inline in the JSX